pallets/time-release/src/lib.rs

```diff
@@ -430,11 +445,11 @@ impl<T: Config> Pallet<T> {
 	}
 
 	fn update_lock(who: &T::AccountId, locked: BalanceOf<T>) {
-		T::Currency::set_lock(RELEASE_LOCK_ID, who, locked, WithdrawReasons::all());
+		let _ = T::Currency::set_freeze(&FreezeReason::NotYetVested.into(), who, locked);
 	}
 
 	fn delete_lock(who: &T::AccountId) {
-		T::Currency::remove_lock(RELEASE_LOCK_ID, who);
+		let _ = T::Currency::thaw(&FreezeReason::NotYetVested.into(), who);
```

Comment:
Let's consider adding an integration test for this and a unit test. We might also be silently failing.